### PR TITLE
Design Picker: Dedupe the themes

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/10110

## Proposed Changes

* Demo of https://github.com/Automattic/dotcom-forge/issues/10110

## Why are these changes being made?

*

## Testing Instructions

* Create a new site
* On the Goals screen, select any goals
* On the Design Picker screen
  * Select any categories
  * Ensure a design won't be shown on multiple categories
  * Ensure a design is shown on the top matched category

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/design-picker/src/hooks/use-filtered-designs.ts [hunk 1]
```diff
@@ -28,21 +28,33 @@ export const getFilteredDesignsByCategory = (
...
 			if ( categorySlugsSet.has( category.slug ) ) {
-				filteredDesignsByCategory[ category.slug ].push( design );
-				count++;
+				// We show the designs for the last selected category on top first
+				// so it would be better to put the design into the last matched category
+				// if it doesn't match all selected categories.
+				lastMatchedCategory = category.slug;
+				matchedCount++;
 			}
 		}
```
Comment (okmttdhr) [inline packages/design-picker/src/hooks/use-filtered-designs.ts]: super badge: slightly improved version in performance for variable assignments 

```js
    const matchedCategories = design.categories.filter(category =>
        categorySlugsSet.has(category.slug)
    );
    const matchedCount = matchedCategories.length;

	...

    if (matchedCategories.length > 0) {
        const lastMatchedCategory = matchedCategories[matchedCategories.length - 1].slug;
        filteredDesignsByCategory[lastMatchedCategory].push(design);
    }
```

**Review by okmttdhr — APPROVED** [review]
Worked well ✅

File: packages/design-picker/src/hooks/use-filtered-designs.ts (hunk 1 above)
Comment (arthur791004) [reply packages/design-picker/src/hooks/use-filtered-designs.ts]: 👍

**Review by fditrapani — APPROVED** [review]
Nice work! looks great.

**Pull request merged**